openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/GlossaryTermRepository.java

```diff
@@ -225,6 +228,35 @@ protected void updateTags(String fqn, String fieldName, List<TagLabel> origTags,
       }
     }
 
+    private void updateChildName(EntityReference child, String parentFqn) throws IOException {
+      GlossaryTerm childTerm = get(null, child.getId(), new Fields(List.of("parent", "children")));
+      String oldFqn = childTerm.getFullyQualifiedName();
+      EntityReference parent = childTerm.getParent();
+      parent.setFullyQualifiedName(parentFqn);
+      setFullyQualifiedName(childTerm);
+      String newFqn = childTerm.getFullyQualifiedName();
+      dao.update(childTerm.getId(), JsonUtils.pojoToJson(childTerm));
+      // update all the tag usages
+      daoCollection.tagUsageDAO().updateTagFqn(oldFqn, newFqn);
+      List<EntityReference> grandChildren = childTerm.getChildren();
+      if (!CollectionUtils.isEmpty(grandChildren)) {
+        for (EntityReference grandChild : grandChildren) {
+          updateChildName(grandChild, newFqn);
+        }
+      }
+    }
+
+    private void updateName(GlossaryTerm origTerm, GlossaryTerm updatedTerm) throws IOException {
+      if (updatedTerm != null && !updatedTerm.getName().equals(origTerm.getName())) {
+        recordChange("name", origTerm.getName(), updatedTerm.getName());
+        List<EntityReference> children = getChildren(updatedTerm);
+        for (EntityReference child : children) {
+          updateChildName(child, updatedTerm.getFullyQualifiedName());
+        }
+        daoCollection.tagUsageDAO().updateTagFqn(origTerm.getFullyQualifiedName(), updatedTerm.getFullyQualifiedName());
```

Comment:
This should generate a change event

Reply:
If a root glossaryTerm has lots of children and grand children terms, updating its name will then create a change event and activity feed for all the assets that are tagged with any term in the lineage. Do we want to create change events recursively for all the affected entities while updating a glossary term?

Reply:
no just the change event for that particular term where the name is being changed.

Reply:
Got it. This patch will do exactly that.